[NO GBP]Fixes space heaters. Makes their effectiveness independant of cell defines.

Pickle-Coding opened the pull request:
## About The Pull Request
Makes BASE_HEATING_ENERGY be 40 KILO JOULES. It shouldn't have any scaling with the cell defines because it impacts how powerful the space heater interacts with the air, and doesn't impact how much energy is actually used per energy it changed in the air.

Makes the efficiency variable inversely scale with STANDARD_CELL_CHARGE. This makes the consumption relative to the standard cell charge, so changing the defines won't impact how long the space heater lasts.

Makes the heating_energy var use initial() instead of the defines, so they respect subtypes better. Makes the improvised chemical heater also scale their variables based on the initial one. Also sets the improvised chemical heater's settable_temperature_range to what it would be set to with tier 1 parts in RefreshParts().

Renames chem_heating_power to beaker_conduction_power to more accurately describe its effect of scaling how fast it heats or cools things. Removes the pointless multiplication that is done to it in every instance of using it, and instead just scale the variable down.

Gives a more accurate description of the improvised chemical heater's heating power in its heating_examine().

Gives better feedback with balloon alerts when attempting to turn on a space heater while it is lacking a cell, has no charge or is broken.
## Why It's Good For The Game
Closes #84892 

This will make their relative energy usage behave identical to before the power changes. It will make their effectiveness independant of the cell scale. Also the improvised chemical heater's heating power description was misleading and nonsensical. Gives a more accurate one. Space heaters give better feedback when attempting to turn it on while it's missing a cell, has no charge or is broken.
## Changelog
:cl:
fix: Fixes space heater cell relative consumption inadvertently being changed from the introduction of megacells.
qol: The improvised chemical heater gives a more accurate description of its heating power on examine.
qol: Improved feedback when attempting to turn on a space heater that is lacking a cell, has no charge or is broken.
code: Space heater relative cell consumption is consistent regardless of the cell charge scale.
/:cl:

Pickle-Coding committed:
Fixes space heaters. Make them scale with cell defines properly.

Makes BASE_HEATING_ENERGY be 40 KILO JOULES. It shouldn't have any scaling with the cell defines because it impacts how powerful the space heater interacts with the air, and doesn't impact how much energy is actually used per energy it changed in the air.

Makes the efficiency variable inversely scale with STANDARD_CELL_CHARGE. This makes the consumption relative to the standard cell charge, so changing the defines won't impact how long the space heater lasts.

Makes the heating_energy var use initial() instead of the defines, so they respect subtypes better. Makes the improvised chemical heater also scale their variables based on the initial one. Also sets the improvised chemical heater's settable_temperature_range to what it would be set to with tier 1 parts in RefreshParts().

Renames chem_heating_power to beaker_conduction_power to more accurately describe its effect of scaling how fast it heats or cools things. Removes the pointless multiplication that is done to it in every instance of using it, and instead just scale the variable down.

Gives a more accurate description of the improvised chemical heater's heating power in its heating_examine().

Pickle-Coding committed:
Improves balloon alerts.

Empty commit because I forgot to include this in the previous commit.

Improved balloon alerts for the user when they attempt to turn on the space heater when it's lacking a cell, has no charge or is broken.

Pickle-Coding committed:
Removes erroneous comma.

Removes an erroneous comma in RefreshParts(). Maybe I should compile after doing a quick one line edit.

Pickle-Coding committed:
More accurate examine.

It's basically just dE/dt = -kE, so we should use a log scale for the display since that's an exponential function.

Pickle-Coding committed:
Stupid.

I was thinking of doing >=, but I ended up with < but forgot to remove the =.

Inari-Whitebear commented:
Ah nice, I was considering putting one up to follow your comments in the issue, but you did it much better than I would've anyway xD

SyncIt21 commented on code/game/machinery/spaceheater.dm:
```diff
@@ -295,7 +295,14 @@
 	mode = HEATER_MODE_STANDBY
 	if(!isnull(user))
-		balloon_alert(user, "turned [on ? "on" : "off"]")
+		if(!cell || QDELETED(cell))
```
```suggestion
		if(QDELETED(cell))
```

it does a null check for us

Pickle-Coding committed:
Removes redundant check.

QDELETED() does a null check for us, so lets remove the additional null check.

Co-authored-by: SyncIt21 <110812394+SyncIt21@users.noreply.github.com>

Pickle-Coding committed:
Fix efficiency of improvised chem heater.

It had the same efficiency as the space heater. I just missed that the 10 was a result of multiplying it by 0.5.

Pickle-Coding committed:
Merge branch 'the-trillionth-spaceheater-pr-this-year' of https://github.com/Pickle-Coding/tgstation into the-trillionth-spaceheater-pr-this-year

Pickle-Coding committed:
The most accurate description.

I forgot that the conduction_modifier gets multiplied by seconds_per_tick instead of using an exponential function. Corrects the heating_examine(), accounting for the timestep inaccuracies, to get complete accuracy with calculating the true conduction power.

Pickle-Coding committed:
Simplifies the math.

Pickle-Coding committed:
Set our_subsystem before parent initialize call.

Sets our_subsystem before the parent Initialize() gets called, so RefreshParts can access it.
